public/service-worker.js

```diff
@@ -277,10 +278,15 @@ self.addEventListener("fetch", (event) => {
 
     // Tries to serve the requested data from the DATA_CACHE
     // If unable to, then it makes a network fetch and stores the fetched data in the cache
+
+    // ML: Why would this third branch ever be taken? why would going to /bookmarks_in_pipeline not be in "navigate mode"
```

Comment:
This handles requests to the /bookmarks_in_pipeline endpoint, which is used to cache the user's exercise words. This fetch occurs whenever the user clicks on the "words" tab and since this request does not change the URL it isn't in "navigate mode".